dev/AppLifecycle/AppLifecycle.cpp

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include <pch.h>
+#include <AppLifecycle.h>
+#include <AppLifecycle.g.cpp>
+
+#include "LaunchActivatedEventArgs.h"
+#include "ProtocolActivatedEventArgs.h"
+#include "Shared.h"
+
+namespace winrt::Microsoft::ProjectReunion::implementation
+{
+    std::tuple<std::wstring, std::wstring> AppLifecycle::ParseCommandLine(std::wstring commandLine)
+    {
+        auto argsStart = commandLine.rfind(L"----") + 4;
```

Comment:
`rfind` and `find_first_of` returns `npos` if the thing isn't there.  Consider adding checks for npos before doing additional math.

Reply:
Updated on latest iteration.